next/src/app/blog/[slug]/page.tsx

```diff
@@ -0,0 +1,123 @@
+import { notFound } from 'next/navigation';
+import Hero from '@/components/_article/hero';
+import AuthorComponent from '@/components/_global/authorComponent';
+import ContentSection from '@/components/_global/contentComponent/contentComponent';
+import SEO from '@/global/Seo';
+import { type BlogPostPage } from '@/types/_pages/BlogPostPage';
+import { sanityFetch } from '@/utils/sanity-client';
+
+export async function generateStaticParams() {
+  const allBlogPosts = await getAllBlogPosts();
+  return allBlogPosts.map(({ slug }) => ({
+    slug: slug.current,
+  }));
+}
+
+export async function generateMetadata({ params: { slug } }: { params: { slug: string } }) {
+  const { seo } = await getMetadata(slug);
+  return SEO({
+    title: seo?.title,
+    description: seo?.description,
+    url: '/co-robie',
+  });
+}
+
+export default async function blogPostPage({ params: { slug } }: { params: { slug: string } }) {
+  const { author, categories, content, hero_Description, hero_Image, hero_Title } = await getBlogData(slug);
+  return (
+    <>
+      <Hero data={{ hero_Description, hero_Image, hero_Title, categories }} />
+      <ContentSection data={content} />
+      <AuthorComponent data={author} />
+    </>
+  );
+}
+
+async function getAllBlogPosts() {
+  const { blogPosts } = await sanityFetch<BlogPostPage>({
+    query: /* groq */ `{
+      "blogPosts": *[_type=="blog_entries"][] {
+        slug {
+          current
+        }
+      }
+    }`,
+  });
+  return blogPosts;
+}
+
+async function getMetadata(slug: string) {
+  const { page } = await sanityFetch<BlogPostPage>({
+    query: /* groq */ `{
+      "page": *[_type=="blog_entries" && slug.current==$slug][0] {
+      seo {
+       ...,
+       og_Img {
+         asset -> {
+           url
+         }
+       }
+     }
+   }
+    }`,
+    params: { slug },
+  });
+  return page;
+}
+
+async function getBlogData(slug: string) {
+  const { page } = await sanityFetch<BlogPostPage>({
+    query: /* groq */ `{
+      "page": *[_type=="blog_entries" && slug.current==$slug][0] {
+        hero_Image {
+          asset -> {
+            altText,
+              url,
+              metadata {
+                lqip,
+                dimensions {
+                  height,
+                  width
+                }
+              }
+          }
+        },
+        hero_Title,
+        content[] {
+          ...
```

Comment:
Don't use merge operator anywhere in query. Explicitly declare them. See: https://www.sanity.io/docs/high-performance-groq#9b1f0b9a4b19. Globally change that.